src/utils/miscellaneous.ts

```diff
@@ -110,7 +110,7 @@ interface RetryOptions {
  * @param exponentialBackOff Whether to use exponential back off, doubling wait interval. Defaults to true
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export async function retry<T extends () => any>(fn: T, options?: RetryOptions): Promise<ReturnType<T>> {
+export async function retry<T extends () => any>(fn: T, options?: RetryOptions): Promise<Unpromise<ReturnType<T>>> {
```

Comment:
Great solution to the nested promises. I didn't know about the `infer` in TS.

`Promise<Unpromise<ReturnType<T>>>`  it's a bit verbose, is not possible to declarare a type like `FlatPromise<ReturnType<T>>>` right?

Reply:
Yes and no <sub><sub>I love this kind of answer</sub></sub>
Would have to do
```ts
export type FlatPromise<T> = Promise<Unpromise<T>>
export const FlatPromise = Promise
```
[because of reasongs](https://github.com/Microsoft/TypeScript/issues/12776)
I don't want to make this hack even more hacky